### PR TITLE
feat(helper): add vue 2 submodule

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This adds `@storyblok/field-plugin/vue2` submodule.

Related:
- https://github.com/storyblok/field-plugin/pull/228
- https://github.com/storyblok/field-plugin/pull/224

## Why?

JIRA: EXT-1828

### CODE REVIEW

**Comment by eunjae-lee** [comment]
If I run this command,
```
yarn workspace field-plugin-vue2-template build
```

I get this error.

```
"default" is not exported by "../../../../node_modules/vue/dist/vue.runtime.esm-bundler.js", imported by "../../../field-plugin/dist/vue2/index.js".
file: /Users/eunjae/workspace/field-plugin/packages/field-plugin/dist/vue2/index.js:2:7
1: import { createFieldPlugin as g } from "@storyblok/field-plugin";
2: import s from "vue";
          ^
3: function m(n, r, _, p, a, l, d, v) {
4:   var e = typeof n == "function" ? n.options : n;
error during build:
RollupError: "default" is not exported by "../../../../node_modules/vue/dist/vue.runtime.esm-bundler.js", imported by "../../../field-plugin/dist/vue2/index.js".
    at error (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:2245:30)
    at Module.error (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:13584:16)
    at Module.traceVariable (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:14009:29)
    at ModuleScope.findVariable (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:12527:39)
    at MemberExpression.bind (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:9462:49)
    at CallExpression.bind (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:5853:23)
    at CallExpression.bind (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:9813:15)
    at VariableDeclarator.bind (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:5853:23)
    at VariableDeclaration.bind (file:///Users/eunjae/workspace/field-plugin/node_modules/rollup/dist/es/shared/node-entry.js:5849:28)
```

When I import the vue 2 helper

```js
import { FieldPluginProvider } from '@storyblok/field-plugin/vue2'
```

this `vue2` submodule has code that imports from vue2, but that's causing this issue, but I'm not sure why, yet.

My theory: when vite imports the submodule, it tries to resolve `vue`, but it's not a dependency of `@storyblok/field-plugin`, so it fails. But... this vue 2 template includes `vue` 🤷🏻‍♂️

One thing I tried is, to copy `helper-vue2/dist/index.js` and bring it directly inside the vue 2 template. It worked. So it's definitely related to how it's importing the library.

WIP

**Comment by eunjae-lee** [comment]
Current dependencies on/for this PR:
* main
  * **PR #231** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/231" target="_blank">Graphite</a>  👈
    * **PR #233** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/233" target="_blank">Graphite</a> 
      * **PR #234** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/234" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/231?utm_source=stack-comment).

**Review by johannes-lindgren — CHANGES_REQUESTED** [review]
This new method (of including field-plugin/react etc)  makes it more complicated to add new templates for new frontend frameworks. Could we describe these added steps in a file, such as `CONTRIBUTING.md`? 

In production, I get the following error: 

> get_html?uid=-preview&protocol=https:&host=app.storyblok.com&preview=1:380 Uncaught ReferenceError: process is not defined

From this line

```js
    productionTip: process.env.NODE_ENV !== 'production',
```

File: packages/cli/templates/vue2/src/App.vue [hunk 1]
```diff
@@ -20,3 +20,9 @@ export default {
   },
 }
 </script>
+
+<style scoped>
+.field-plugin-provider {
```
[the remark this answers is not in the document]
Comment (eunjae-lee) [reply packages/cli/templates/vue2/src/App.vue:25]: I thought so, too. But it works! 
https://vuejs.org/api/sfc-css-features.html#child-component-root-elements

**Review by johannes-lindgren — APPROVED** [review]

**Comment by eunjae-lee** [comment]
@eunjae-lee started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/231).

**Comment by eunjae-lee** [comment]
@eunjae-lee merged this pull request with [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/231).

**Pull request merged**